stix2/datastore/taxii.py

```diff
@@ -145,9 +179,12 @@ def get(self, stix_id, version=None, _composite_filters=None):
             stix_objs = self.collection.get_object(stix_id)["objects"]
             stix_obj = list(apply_common_filters(stix_objs, query))
 
-        except HTTPError:
-            # if resource not found or access is denied from TAXII server, return None
-            stix_obj = []
+        except HTTPError as e:
+            if e.response.status_code == 404:
```

Comment:
Would it make sense to add a `logging` statement here. I was looking back to the related issues and @johnwunder was specifically confused about 404 returning an empty list. I know raising the exception is unwanted so maybe logging would at least make you aware that an 404 occured... Any thoughts?

Reply:
Ill yield to @gtback on the logging. But again, the logic implemented is pulled from the policy outlined here https://github.com/oasis-open/cti-python-stix2/issues/113.

Reply:
Hmmm... didn't he try a "query()" (plural getter, see my other comment below) on a non-existent collection?  I think that would result in an exception now.  Although I'm never averse to having useful diagnostics :)